05-onchain.md

```diff
@@ -305,16 +305,19 @@ There are several possible cases for an offered HTLC:
    HTLC. In this case, the recipient must use the preimage, once it receives it
    from the outgoing HTLC; otherwise, it will lose funds by sending an outgoing
    payment without redeeming the incoming payment.
+4. The HTLC is forwarded as above, but the recipient also happens to be the
+   final destination for the payment.  In this case, it knows the preimage but
+   as it has not received the full payment, it must not reveal
+   it.<sup>[Preimage-Extraction](https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-October/002857.html)</sup>.
```

Comment:
I'm not sure what I offer is better, but the current wording is a bit hard to understand (at least for me).

```suggestion
4. The HTLC is forwarded as above, but the recipient also happens to be the
   final destination for the payment.  In this case, it knows the preimage but
   until the corresponding outgoing HTLC is settled, it must not reveal
   it.<sup>[Preimage-Extraction](https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-October/002857.html)</sup>.
```